pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java

```diff
@@ -119,19 +120,22 @@ public void run() throws Exception {
 
     FileSystem fs = FileSystem.get(getConf());
     Path inputPathPattern = new Path(_inputSegmentDir);
+    Path stagingDir = new Path(_stagingDir);
+    Path outputDir = new Path(_outputDir);
 
-    if (fs.exists(new Path(_stagingDir))) {
+    if (fs.exists(stagingDir)) {
       LOGGER.warn("Found the temp folder, deleting it");
-      fs.delete(new Path(_stagingDir), true);
+      fs.delete(stagingDir, true);
     }
-    fs.mkdirs(new Path(_stagingDir));
+    fs.mkdirs(stagingDir);
+    FileSystem.get(getConf()).setPermission(stagingDir, new FsPermission("755"));
```

Comment:
We should take the FsPermission as an argument, which either comes from a config or defaults to permission of the parent of destination (where the temp is moved to).